index.js

```diff
@@ -1330,59 +1378,99 @@ function checkCode(context) {
-                                }
-                                if (!words[word].ru) {
-                                    context.errors.push(`[E506] Word "${word}" is not translated to russian in admin/words.js. You can use https://translator.iobroker.in/ for translations`);
-                                }
-                            });
+                if (context.ioPackageJson.common.adminUI && context.ioPackageJson.common.adminUI.config === 'json') {
+                    if (context['/admin/jsonConfig.json']) {
+                        try {
+                            JSON.parse(context['/admin/jsonConfig.json']);
+                        } catch (e) {
+                            context.errors.push('[E507] Cannot parse "admin/jsonConfig.json": ' + e);
                         }
+                    } else {
+                        context.errors.push(`[E508] "admin/jsonConfig.json" not found, but admin support is declared`);
+                    }
+
+                    if (context['/admin/i18n/en/translations.json']) {
+                        try {
+                            JSON.parse(context['/admin/i18n/en/translations.json']);
+                        } catch (e) {
+                            context.errors.push('[E509] Cannot parse "admin/i18n/en/translations.json": ' + e);
+                        }
+                    } else {
+                        context.errors.push(`[E510] "/admin/i18n/en/translations.json" not found, but admin support is declared`);
                     }
                 }
 
-                if (context.ioPackageJson.common &&
-                    ((context.ioPackageJson.common.adminUI && context.ioPackageJson.common.adminUI.custom === 'json') ||
-                        context.ioPackageJson.common.jsonCustom)) {
-                    if (!context['/admin/jsonCustom.json']) {
-                        context.errors.push(`[E506] "admin/jsonCustom.json" not found, but custom support is declared`);
+                if (context.ioPackageJson.common.jsonCustom || (context.ioPackageJson.common.adminUI && context.ioPackageJson.common.adminUI.custom === 'json')) {
+                    if (context['/admin/jsonCustom.json']) {
+                        try {
+                            JSON.parse(context['/admin/jsonCustom.json']);
+                        } catch (e) {
+                            context.errors.push('[E511] Cannot parse "admin/jsonCustom.json": ' + e);
+                        }
+                    } else {
+                        context.errors.push(`[E512] "admin/jsonCustom.json" not found, but custom support is declared`);
                     }
                 }
 
+                if (context['/iob_npm.done']) {
+                    context.errors.push('[E503] "iob_npm.done" found in repo! Please add it to .gitignore');
```

Comment:
Better "please remove this file"

Reply:
I have not idea what that file is used for (that rule already existed before this PR)

Reply:
When you install an adapter in your iobroker system then iobroker creates that file in the dir after installation ... or better "Older versions of controller". And some developers dvelop the adapter directly in iorboekr and then copy all stuff over ... this means that they also copy that file ... thats why we check it and tell it is deleted